src/merkle_tree.rs

```diff
@@ -0,0 +1,470 @@
+use lazy_static::lazy_static;
+use std::vec;
+
+// The hash of the zero value
+lazy_static! {
+    static ref DEFAULT_ZERO_HASH: String = sha256::digest(&[0]);
+}
+
+type Input = Vec<Vec<u8>>;
+type HashTree = Vec<Vec<String>>;
+
+#[derive(Debug, PartialEq)]
+pub enum MerkleTreeError {
+    DataLengthIsZero,
+    DataIsEmpty,
+}
+
+pub struct MerkleTree {
+    merkle_tree: HashTree,
+    last_element_index: usize,
+}
+impl MerkleTree {
+    /// Create a default Merkle Tree
+    fn default() -> Self {
+        Self {
+            merkle_tree: Vec::new(),
+            last_element_index: 0,
+        }
+    }
+    /// Returns the height of the Merkle Tree, meaning the amount of levels.
+    pub fn height(&self) -> usize {
+        self.merkle_tree.len()
+    }
+
+    /// Returns the root of the Merkle Tree.
+    pub fn get_root(&self) -> String {
+        self.get_tree_element(0, 0).clone()
+    }
+
+    /// Returns an immutable reference to the hash stored in a certain level and index
+    /// of the Merkle Tree.
+    pub fn get_tree_element(&self, level: usize, index: usize) -> &String {
+        &self.merkle_tree[level][index]
+    }
+
+    /// Returns the index of the last leaf with real data.
+    pub fn last_element_index(&self) -> usize {
+        self.last_element_index
+    }
+
+    /// Returns an immutable reference to an entire level of the tree.
+    pub fn get_level(&self, level: usize) -> &Vec<String> {
```

Comment:
It's better to use `&[T]` in place of `&Vec<T>` whenever possible. Using `Vec` exposes internal details without any additional benefits.

```suggestion
    pub fn get_level(&self, level: usize) -> &[String] {
```

Reply:
Thank you :) Applied it [here](https://github.com/sofiazcoaga/rusty-merkle-tree/pull/16/commits/7e14d209841fa0e9d97cab73d821857e39262825)